bsp/Src/UART.c

```diff
@@ -11,13 +11,13 @@ typedef struct {
     uint8_t data[DATA_SIZE]; // data to be transmitted, 1 byte
 } tx_payload_t;
 
-uint16_t sizeof_tx_payload_t = sizeof(tx_payload_t); // Precompute size of payload
+static uint16_t sizeof_tx_payload_t = sizeof(tx_payload_t); // Precompute size of payload
```

Comment:
Another thing about this whole "precomputed size of payload" is that we don't really have to allocate space for this being a new variable.

sizeof is a preprocessing step, so if you just use `sizeof(tx_payload_t)` wherever you want it will be replaced by the compiler with a number. so it's not really necessary to make these variables in the first place even.